app/src/main/java/org/oppia/app/player/audio/AudioFragmentPresenter.kt

```diff
@@ -84,10 +99,46 @@ class AudioFragmentPresenter @Inject constructor(
       it.audioFragment = fragment as AudioFragment
       it.lifecycleOwner = fragment
     }
-
+    subscribeToProfileLiveData()
     return binding.root
   }
 
+  private val profileLiveData: LiveData<Profile> by lazy {
+    getProfileData()
+  }
+
+  private fun getProfileData(): LiveData<Profile> {
+    return Transformations.map(
+      profileManagementController.getProfile(profileId),
+      ::processGetProfileResult
+    )
+  }
+
+  private fun subscribeToProfileLiveData() {
+    profileLiveData.observe(activity, Observer<Profile> { result ->
+      audioLanguage = result.audioLanguage
+      languageSelected(getAudioLanguage(audioLanguage))
+    })
+  }
+
+  /** Gets language code by [AudioLanguage]. */
+  fun getAudioLanguage(audioLanguage: AudioLanguage): String {
+    return when (audioLanguage) {
+      AudioLanguage.NO_AUDIO -> "No Audio"
+      AudioLanguage.ENGLISH_AUDIO_LANGUAGE -> "en"
+      AudioLanguage.HINDI_AUDIO_LANGUAGE -> "hi"
+      AudioLanguage.FRENCH_AUDIO_LANGUAGE -> "fr"
+      AudioLanguage.CHINESE_AUDIO_LANGUAGE -> "zh"
+      else -> "No Audio"
+    }
+  }
+
+  private fun processGetProfileResult(profileResult: AsyncResult<Profile>): Profile {
+    if (profileResult.isFailure()) {
```

Comment:
Probably related to the above comment, but please handle the error case here.

Reply:
We removed "No Audio" from options default options and also added english as default value.

Reply:
Thanks--I added another comment that is a bit more specific with the most recent changes.